### PR TITLE
Issue #7: NewlineAtEndOfFile, RegexpOnFilename, FileLength's context strategy for SuppressionPatchFilter

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #7: NewlineAtEndOfFile, RegexpOnFilename, FileLength's context strategy for SuppressionPatchFilter

- FileLength
- RegexpOnFilename
- NewlineAtEndOfFile

Considering that these violations are only one in one file, so we can exactly captured them through when we know the AuditEvents are created by these checks, only using `isFileNameMatching` in patch-filter not need `isLineMatching`'s line comparison.

This PR is based on https://github.com/checkstyle/patch-filters/pull/96, please review https://github.com/checkstyle/patch-filters/pull/96 first.

This PR also can solve https://github.com/checkstyle/patch-filters/issues/104

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java [hunk 1]
```diff
@@ -74,13 +74,22 @@ private boolean isFileNameMatching(AuditEvent event) {
...
-                    break;
+            final String[] checkNames = event.getLocalizedMessage().getSourceName().split("\\.");
+            final String checkShortName = checkNames[checkNames.length - 1];
+            if ("RegexpOnFilenameCheck".equals(checkShortName)
+                    || "FileLengthCheck".equals(checkShortName)
+                    || "NewlineAtEndOfFileCheck".equals(checkShortName)) {
+                result = true;
+            }
+            else {
+                for (List<Integer> aLineRangeList : lineRangeList) {
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java:85]: you can use stream api here

Smth like
```
return aLineRangeList.stream().any(lineRange -> {...})
```

Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java:85]: @timurt 
Is like this one, but I think it is less readable.
```java
private boolean isLineMatching(AuditEvent event) {
        boolean result = false;
        if (event.getLocalizedMessage() != null) {
            for (List<Integer> aLineRangeList : lineRangeList) {
                final int currentLine = event.getLine();
               result =
                       (aLineRangeList.stream().anyMatch(position -> currentLine >= position + 1 )
                           && aLineRangeList.stream().anyMatch(position -> currentLine < position + 1));
//                final int startLine = aLineRangeList.get(0) + 1;
//                final int endLine = aLineRangeList.get(1) + 1;
//                result = currentLine >= startLine && currentLine < endLine;
                if (result) {
                    break;
                }
            }
        }
        return result;
    }
```

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java [hunk 2]
```diff
@@ -74,13 +74,22 @@ private boolean isFileNameMatching(AuditEvent event) {
...
-            for (List<Integer> aLineRangeList : lineRangeList) {
-                final int startLine = aLineRangeList.get(0) + 1;
-                final int endLine = aLineRangeList.get(1) + 1;
-                final int currentLine = event.getLine();
-                result = currentLine >= startLine && currentLine < endLine;
-                if (result) {
-                    break;
+            final String[] checkNames = event.getLocalizedMessage().getSourceName().split("\\.");
+            final String checkShortName = checkNames[checkNames.length - 1];
+            if ("RegexpOnFilenameCheck".equals(checkShortName)
```
[the remark this answers is not in the document]
Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java:79]: done

**Comment by rdiachenko** [comment]
@HuGanghui please resolve a conflict.

**Pull request merged**